pkg/registry/flowcontrol/ensurer/strategy.go

```diff
@@ -267,36 +269,36 @@ func EnsureConfiguration[ObjectType configurationObjectType](ctx context.Context
 
 		// we always re-create a missing configuration object
 		if _, err = ops.Create(ctx, ops.DeepCopy(bootstrap), metav1.CreateOptions{FieldManager: fieldManager}); err == nil {
-			klog.V(2).InfoS(fmt.Sprintf("Successfully created %T", bootstrap), "type", configurationType, "name", name)
+			logger.V(2).Info(fmt.Sprintf("Successfully created %T", bootstrap), "type", configurationType, "name", name)
 			return nil
 		}
 
 		if !apierrors.IsAlreadyExists(err) {
 			return fmt.Errorf("cannot create %T type=%s name=%q error=%w", bootstrap, configurationType, name, err)
 		}
-		klog.V(5).InfoS(fmt.Sprintf("Something created the %T concurrently", bootstrap), "type", configurationType, "name", name)
+		logger.V(5).Info(fmt.Sprintf("Something created the %T concurrently", bootstrap), "type", configurationType, "name", name)
 	}
 
-	klog.V(5).InfoS(fmt.Sprintf("The %T already exists, checking whether it is up to date", bootstrap), "type", configurationType, "name", name)
+	logger.V(5).Info(fmt.Sprintf("The %T already exists, checking whether it is up to date", bootstrap), "type", configurationType, "name", name)
 	newObject, update, err := strategy.ReviseIfNeeded(ops, current, bootstrap)
 	if err != nil {
 		return fmt.Errorf("failed to determine whether auto-update is required for %T type=%s name=%q error=%w", bootstrap, configurationType, name, err)
 	}
 	if !update {
-		if klogV := klog.V(5); klogV.Enabled() {
-			klogV.InfoS("No update required", "wrapper", bootstrap.GetObjectKind().GroupVersionKind().Kind, "type", configurationType, "name", name,
+		if v5 := logger.V(5); v5.Enabled() {
+			v5.Info("No update required", "wrapper", bootstrap.GetObjectKind().GroupVersionKind().Kind, "type", configurationType, "name", name,
 				"diff", cmp.Diff(current, bootstrap))
 		}
 		return nil
 	}
 
-	if _, err = ops.Update(ctx, newObject, metav1.UpdateOptions{FieldManager: fieldManager}); err == nil {
-		klog.V(2).Infof("Updated the %T type=%s name=%q diff: %s", bootstrap, configurationType, name, cmp.Diff(current, bootstrap))
+	if updated, err := ops.Update(ctx, newObject, metav1.UpdateOptions{FieldManager: fieldManager}); err == nil {
+		logger.V(2).Info(fmt.Sprintf("Updated the %T", bootstrap), "type", configurationType, "name", name, "diff", cmp.Diff(current, updated))
```

Comment:
It was this way before, but let me ask anyway as this gets touched here: would it make sense to only execute `cmd.Diff` if the log message really gets executed? In other words, how large are `current` and `updated`?

There's no such helper currently available, but one could be added so that one could write `klog.Stringer(func() string { return cmp.Diff(current, updated) })`.

Reply:
The same could be used to avoid the `Sprintf`:
```
Info("Updated bootstrap", "goType", klog.Stringer(func() string { fmt.Sprintf("%T", bootstrap) }), ...
```

The exact message is up for debate. "Updated object"?

Reply:
I left this as it is because updates won't happen often, and if they do it's better to see them.

There is another topics around using go-cmp in the first place: https://github.com/kubernetes/kubernetes/issues/104821. @s-urbaniak is looking into some of the cases where we misuse go-cmp in production code today.